duckduckgo/macos-browser
# Connection interruption simulation option

Task/Issue URL: https://app.asana.com/0/0/1205601804599806/f
BSK PR: https://github.com/duckduckgo/BrowserServicesKit/pull/516

**Description**:

This PR adds a `Connection Interruption` failure simulation option and updates BSK to bring in the changes made to allow this.

I also added some changes to the PacketTunnelProvider in order to test connection interruptions for the iOS NetP Notifications project. I found that simply setting reasserting to true causes the ConnectionnTester to trigger an interruption, then setting it to false recovers it. This has the benefit of testing the result of the real callback, not needing to mess with any of the connection tester’s callback handling, not needing to add any new state and also mimicking an actual interruption (as the VPN connection status also changes).

**Steps to test this PR**:
1. Build this branch through Xcode
2. Start Network Protection
3. If this is the first time launching, agree to any notifications prompt.
4. Go to the Debug menu and simulate a connection interruption (Debug -> Network Protection -> Simulate Failure -> Connection Interruption)
5. **Observe the interruption notification**
6. Go back to Network Protection Status View
7. **Observe the status is reconnecting**
8. Wait for the reconnected notification.

—
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
